mkapp/mkapp_ota.sh

```diff
@@ -1,128 +1,39 @@
-	${BIN_DIR}/update_mbr sys_partition_tmp_app.bin 1 sunxi_mbr_tmp_app.fex
+BIN_DIR=$ROOT_DIR/bin
+IMG_DIR=$ROOT_DIR/ota_app
+APP_DIR=$ROOT_DIR/app
 
-	local APP_PART_DOWNLOAD_FILE_VER=app.ver
-	local APP_PART_DOWNLOAD_FILE=app.fex
-	local APP_PART_FILE_PATH=$ROOT_DIR/app
-	local APP_PART_SIZE_IN_SECTOR=`${BIN_DIR}/parser_mbr sunxi_mbr_tmp_app.fex get_size_by_name ${APP_PART_NAME}`
+APP_SIZE=12582912
+APP_IMAGE=${IMG_DIR}/app.fex
+APP_VERSION=$(get_app_version "$ROOT_DIR/app/version")
 
-	if [ x${APP_PART_DOWNLOAD_FILE} != x'' -a  x${APP_PART_SIZE_IN_SECTOR} != x'' ]; then
-		let APP_PART_SIZE_IN_K=$APP_PART_SIZE_IN_SECTOR/2
-		echo "APP_PART_DOWNLOAD_FILE = ${IMG_DIR}/${APP_PART_DOWNLOAD_FILE}"
-		echo "---liran: make_ext4fs -l ${APP_PART_SIZE_IN_K}k -b 1024 -m 0 -j 1024 ${ROOT_DIR}/${APP_PART_DOWNLOAD_FILE} ${APP_PART_FILE_PATH}"
-		rm -f ${IMG_DIR}/${APP_PART_DOWNLOAD_FILE}
-		#${BIN_DIR}/make_ext4fs -l ${APP_PART_SIZE_IN_K}k -b 1024 -m 0 -j 1024 ${IMG_DIR}/${APP_PART_DOWNLOAD_FILE}  ${APP_PART_FILE_PATH}
-		
-		let APP_PART_SIZE_IN_BYTES=${APP_PART_SIZE_IN_K}*1024
-		echo "---liran: mkfs.jffs2 l -e 0x10000 -p ${APP_PART_SIZE_IN_BYTES} -d ${APP_PART_FILE_PATH} -o ${IMG_DIR}/${APP_PART_DOWNLOAD_FILE}"
-		${BIN_DIR}/mkfs.jffs2 -l -e 0x10000 -p ${APP_PART_SIZE_IN_K}k -d ${APP_PART_FILE_PATH} -o ${IMG_DIR}/${APP_PART_DOWNLOAD_FILE}
-		[ -f ${APP_PART_FILE_PATH}/version ] && cp ${APP_PART_FILE_PATH}/version ${IMG_DIR}/${APP_PART_DOWNLOAD_FILE_VER}
-	else
-		printf "no app resource partitions\n"
-	fi
-	cd -
-}
+rm -rf $IMG_DIR
+mkdir -p $IMG_DIR
 
-function get_app_version() 
-{
-	# key和value的分隔符，即等号两边有没有空格
-	#delimeter='='
-	delimeter=' = '
-    file=$1
-	section="app"
-    key="ver"
-    val=$(awk -F "$delimeter" '/\['${section}'\]/{a=1}a==1&&$1~/'${key}'/{print $2;exit}' $file)
-    echo ${val}
-}
-
-function make_ota_package_for_app()
-{
-	local T=$ROOT_DIR
-	local OTA_DIR=$T/ota_app
-	local make_fail=0
-	local ota_version=$T/image/app.ver
-	local VER=$(get_app_version "$ota_version")
-	local ota_package=hdzgoggle_app_ota-${VER}.tar
+${BIN_DIR}/mkfs.jffs2 -l -e 0x10000 \
+	-p ${APP_SIZE} \
+	-d ${APP_DIR} \
+	-o ${APP_IMAGE}
 
-	mkdir -p "$OTA_DIR"
-	\cd "$OTA_DIR"
-	rm -f $ota_package $ota_package.md5
-
-	target_list="$T/image/app.fex $ota_version"
-	for i in $target_list; do
-		img=${i##*/}
-		if [ ! -f "$i" ]; then
-			print_red "$i not exsit!"
-			make_fail=1
-			break
-		fi
-		dd if="$i" of="$img" bs=512 conv=sync
-		make_img_md5 "$img"
-		tar -rvf "$ota_package" "$img"
-		tar -rvf "$ota_package" "$img.md5"
-	done
-	if [ x$make_fail = x"1" ]; then
-		print_red "make fail"
-		rm -f $ota_package
-	else
-		make_img_md5 "$ota_package"
-		print_red "$OTA_DIR/$ota_package"
-	fi
-	\cd - > /dev/null
-}
+make_img_md5 ${APP_IMAGE}
 
-envsetup
-make_app_res $SYS_PARTITION
-make_ota_package_for_app
\ No newline at end of file
+pushd $IMG_DIR
+tar cvf $IMG_DIR/hdzgoggle_app_ota-${APP_VERSION}.tar *
+popd
```

Comment:
Probably better as
```
rm -f $IMG_DIR/hdzgoggle_app_ota-${APP_VERSION}.tar
tar cf $IMG_DIR/hdzgoggle_app_ota-${APP_VERSION}.tar -C $IMG_DIR .
```

Reply:
i had something like that initially, but that leaves us with an ugly leading `./` in the archive, which is a bit of a pet-peeve of mine. Most tools should handle that fine, but hard to tell if that would cause any issues down the line.

Reply:
Fair call. I just don't think the `popd` is required because the script is ending. Which means the `pushd` can just be `cd`.